### ISSUE TITLE
Suggest Alluxio does determine format  intellgiently

### ISSUE THREAD

**Issue by cheyang** [Alluxio/alluxio#9712]
**Problem Summary:**

When running alluxio in Kubernetes, if using `/entrypoint.sh master format` to start alluxio master. There is a highly potential risk to format the meta data the journal disk when alluxio master is restarted. 

But when using `/entrypoint.sh master --no-format` ,  the user has to run `alluxio formatMaster` before starting the alluxio master. I think it's not a good user experience. 

**Suggestion:**

Can we make alluxio to determine if the format is necessary when it's start. For example, if the directory is not empty, the alluxio will not do format by default. if the directory is empty, it can format.

**Comment by rastogiasr** [comment]
Bin will handle this issue.

**Comment by LuQQiu** [comment]
@ssz1997 FYI

**Comment by ssz1997** [comment]
@jenoudet From what I know, if the journal directory is empty, it is unnecessary to run formatMaster/formatJournal, right?

**Comment by jenoudet** [comment]
> @jenoudet From what I know, if the journal directory is empty, it is unnecessary to run formatMaster/formatJournal, right?

That is true for the embedded journal. The UFS journal actually creates a specific structure for its journal files when it formats.
